### PR TITLE
MODE-2360: Add content hash property

### PR DESCRIPTION
**Pull request by okulikov**
This PR introduces cmis:contentStreamHash property

### CODE REVIEW

**Comment by rhauch** [comment]
Looks good to me, as long as that hex hash string conforms to the right CMIS format.

**Comment by okulikov** [comment]
@rhauch , yes hash string follows CMIS format: {hash-alg}hash-string

File: web/modeshape-web-cmis/pom.xml [hunk 1]
```diff
@@ -80,7 +80,6 @@
         <dependency>
             <groupId>org.modeshape</groupId>
             <artifactId>modeshape-jcr</artifactId>
-            <scope>test</scope>
```
Comment (hchiorean) [inline web/modeshape-web-cmis/pom.xml]: @okulikov: the code should be using `org.modeshape.jcr.api.Binary` and then this change would not be needed - i.e. it should still be possible to compile against `modeshape-jcr-api` only

File: web/modeshape-web-cmis/src/main/java/org/modeshape/cmis/JcrMsVersion.java [hunk 2]
```diff
@@ -84,4 +93,28 @@ public JcrVersion getVersion(String name) {
...
+    /**
+     * Get the hexadecimal form of the SHA-1 hash of the contents. 
+     * 
+     * @return the hexadecimal form of hash, or a null string if the hash could not be computed or is not known
+     */
+    private String getHash() {
+        try {
+            Node contentNode = getContextNode();
+            Property data = contentNode.getProperty(Property.JCR_DATA);
+            AbstractBinary bin = (AbstractBinary) data.getBinary();
```
Comment (hchiorean) [inline web/modeshape-web-cmis/src/main/java/org/modeshape/cmis/JcrMsVersion.java]: see my previous comment

**Comment by okulikov** [comment]
@hchiorean , fixed

**Pull request merged**